processor/manager.go

```diff
@@ -35,30 +34,37 @@ type LifecycleManager struct {
 // Start starts a processor, this is not a blocking call.
 // If the processor is not completely started and the data started coming then also it will not be problematic as we
 // are assuming that the DBs will be up.
-func (proc *LifecycleManager) Start() {
+func (proc *LifecycleManager) Start() error {
 	if proc.Transformer != nil {
 		proc.HandleT.transformer = proc.Transformer
 	}
 
-	proc.HandleT.Setup(proc.BackendConfig, proc.gatewayDB, proc.routerDB, proc.batchRouterDB,
-		proc.errDB, proc.clearDB, proc.ReportingI, proc.MultitenantStats, proc.transientSources, proc.rsourcesService)
+	proc.HandleT.Setup(
+		proc.BackendConfig, proc.gatewayDB, proc.routerDB, proc.batchRouterDB, proc.errDB,
+		proc.clearDB, proc.ReportingI, proc.MultitenantStats, proc.transientSources, proc.rsourcesService,
+	)
 
 	currentCtx, cancel := context.WithCancel(context.Background())
 	proc.currentCancel = cancel
 
-	g, ctx := errgroup.WithContext(currentCtx)
-	proc.waitGroup = g
-	g.Go(func() error {
-		proc.HandleT.Start(ctx)
-		return nil
-	})
+	var wg sync.WaitGroup
```

Comment:
Why would we need a errgroup or a waitgroup when we are not waiting for the function to return? Can't we just get rid of the wg here?

Reply:
Have you checked the rest of the changes in this file? Look at the next line (i.e. 51) where I do `proc.waitGroup = &wg`. The `Wait()` is used in the `Stop()` method, I didn't want to change the existing logic and waiting when stopping makes sense to me.

Reply:
Somehow I missed this.. Thanks for the explaining